spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/data/AvroDataTest.java

```diff
@@ -285,8 +282,8 @@ public void testMissingRequiredWithoutDefault() {
                 .build());
 
     assertThatThrownBy(() -> writeAndValidate(writeSchema, expectedSchema))
-        .isInstanceOf(IllegalArgumentException.class)
-        .hasMessage("Missing required field: missing_str");
+        .hasRootCauseInstanceOf(IllegalArgumentException.class)
+        .hasMessageContaining("Missing required field: missing_str");
```

Comment:
This was needed to validate the reader failure in `testMissingRequiredWithoutDefault` in Spark scans because the failure happens on executors and is wrapped in `SparkException` when it is thrown on the driver.